### DISCUSSION TITLE
Any way to cancel a solve.

### DISCUSSION THREAD

**Discussion by AndreasUlven** [coin-or/Clp#198] [Q&A] [answered]
Hello,

I've search through the documentation after a way to cancel the run gracefully between iterations or at other times. Is it possible to do this? Note that I'm not thinking about the max time or max iteration count setting, more in the lines of a flag to set that clp watches or callback functionality that can stop the process.

For example, the user of my program wants to be able to cancel during a solve without resorting to stopping threads or similar. So if there is a way to gracefully cancel for example between iterations that would be great. 

I wondered if you could try to set max iteration count to 0 or 1 in order to stop it but that seems like it could cause some problems, if it's even possible.

**Comment by jhmgoossens** [comment, marked as answer]
Haven’t tried this, but take a look at [ClpEventHandler](https://github.com/coin-or/Clp/blob/master/src/ClpEventHandler.hpp) and the example in MyEventHandler.  Does that help?

**Reply by AndreasUlven** [reply]
This was exactly what I needed, thank you!

**Comment by tkralphs** [comment]
It's not clear to me exactly what you're after, but Clp does have a signal handler, so ctrl-c seems to do what you want. It will exit gracefully and print out partial results.
